middlewares/discountChecker.js

```diff
@@ -0,0 +1,18 @@
+exports.couponOwnerChecker = function (req, res, next) {
+  var username = req.query.username;
+  var orderValue = req.query.orderValue;
+
+  if(!username) {
+  	var err = new Error('No Username Provided');
+    err.status = 406;
+    next(err);
+  }
+
+  if(!orderValue) {
+  	var err = new Error('No orderValue Provided');
+    err.status = 406;
+    next(err);
+  }
+  next();
+};
+
```

Comment:
this middleware validates username and order value however, it's unlikely that these checks will be used by other routers, why not putting the checks into the router?